Fixing some bugs

angelitto2005 opened the pull request:
Hi again,
It looks like I've found another bug.

The Bug:
Currently, the transcribe function can silently fail to transcribe an entire audio segment. This occurs when the Gemini API returns a valid but empty JSON response (e.g., []).
This can happen either because of a transient API error or if the API incorrectly detects no speech in the segment. The current script logic misinterprets this empty response as a "successful" transcription of a silent chunk.
The consequences are:

1. A gap is created in the final .srt transcript.
2. No error is reported to the user. The process simply moves to the next segment.
3. The user is unaware that a segment with audible speech may have been lost.

The Fix:
```

                            transcription_json = json_repair.loads(response_text)
                            
                            # --- PROPOSED MODIFICATION ---
                            # Check if the API returned an empty list. If so, it means no speech
                            # was detected, or there was a transient error. Instead of silently
                            # skipping the segment, we raise an error to trigger a retry.
                            if not transcription_json:
                                raise ValueError("API returned an empty transcript. Retrying segment.")
                            # --- END OF MODIFICATION ---

                            for i in range(len(transcription_json)):
```

This commit introduces a simple check immediately after the API response is parsed in the transcribe function.
    After the line transcription_json = json_repair.loads(response_text), we now check if the resulting list is empty (if not transcription_json:).
    If the list is empty, a ValueError is intentionally raised.
    This error is caught by the existing except block, which prevents the chunk_processed_successfully flag from being set to True.
    Because the flag remains False, the while loop iterates again, forcing the script to retry sending the same audio segment to the API.

This change makes the transcription process more robust and resilient. It ensures that transient API errors resulting in empty responses do not cause audio segments to be silently skipped, preserving the integrity of the final transcript.

Before fix
<img width="1089" height="513" alt="Captură de ecran 2025-08-05 175542" src="https://github.com/user-attachments/assets/0bc12644-41bf-4a73-bc77-b1177a7aebc6" />


After fix
<img width="1025" height="184" alt="Captură de ecran 2025-08-05 202201" src="https://github.com/user-attachments/assets/ca46f18e-dba3-47f9-b01a-6cf548e0058d" />


P.S. After the recent changes in main.py, now it looks like this:

<img width="1022" height="92" alt="Captură de ecran 2025-08-06 175550" src="https://github.com/user-attachments/assets/aeb8c2de-accf-47d6-b2e1-ac1e08bc4a49" />

angelitto2005 committed:
Fixing some bugs

angelitto2005 commented:
### Fix2: Add retry limit for empty responses to prevent infinite loops on silent segments

**The Problem:**
The previous fix correctly retries a transcription when the API returns an empty response. However, this creates an infinite loop if a segment is genuinely silent (e.g., music-only sections at the end of a file). The script will retry indefinitely, as the API will consistently return an empty response for that segment.

**The Solution:**
This commit enhances the retry logic by introducing a counter and a limit.

1. A retry counter (empty_response_retries) is initialized for each audio chunk.
2. When an empty response is detected, the counter is incremented.
3. The script will retry up to a defined maximum (max_empty_retries, e.g., 3 times).
4. If the limit is reached, the script logs a final warning, assumes the segment is permanently untranscribable (likely silent), and breaks the retry loop to proceed to the next segment.


### Fix3: Implement overlapping chunks to fix sentence cutoff in Transcribe.


The previous transcription method processed audio in distinct, sequential chunks. This caused sentences that spanned the boundary of two chunks to be cut in half, leading to their incorrect transcription or complete omission by the AI model.

This commit resolves these boundary artifacts by implementing an overlapping chunk processing strategy, which significantly improves transcription accuracy and reliability.

Key Changes:

    Overlapping Logic: The main processing loop now calculates audio chunks that overlap by a predefined overlap_seconds value. This ensures that any sentence at a boundary is fully contained in the subsequent chunk.

    Duplicate Filtering: To prevent duplicate subtitles from appearing in the final output, the system now tracks the end time of the last validly added subtitle (last_subtitle_end_time). New subtitles from a chunk are only added if they start after this timestamp.

    Robust Progress Loop: The main while loop's progression is now tied to the end of the processed chunk, guaranteeing that the script always moves forward and exits correctly, preventing infinite loops or premature termination.

    Accurate Progress Saving: The resume functionality now saves the last_subtitle_end_time, ensuring that if the script is interrupted, it can resume from the precise end of the last correctly transcribed line, not just the start of the last chunk.


**Fix4: Two progress bar in translation mode**

Although the transcription works fine, in the translation mode, the progress bar appears twice, like this

<img width="851" height="140" alt="Captură de ecran 2025-08-06 170530" src="https://github.com/user-attachments/assets/1cc6b609-97ec-42c6-84ee-2797b399f7b6" />

The problem:
<img width="1012" height="231" alt="Captură de ecran 2025-08-06 170238" src="https://github.com/user-attachments/assets/5cb7f010-cb4e-44a5-8d78-2d60778d1cd1" />

The fix:
<img width="1023" height="233" alt="Captură de ecran 2025-08-06 170350" src="https://github.com/user-attachments/assets/dc2a3509-4f9a-4626-b9a4-8bd36f791b10" />

angelitto2005 commented:
I tried to keep your original main, without my colors or Romanian comments in the file 😊
I modified only `info_with_progress` instead of `info` at lines 587 and 589.
All other changes were made within the `def transcribe` function.

I hope everything is fine now.

And sorry if I make mistakes when submitting a PR, I'm still learning 😊

MaKTaiL commented:
Couldn't this be fixed by changing the system instructions to make sure it never returns an empty object?

angelitto2005 commented:
I don't know if it's possible...
But if it's an error from Gemini, I don't think you can fix it just with instructions.
But I'm not sure. If you have an idea, try it and see what happens.

angelitto2005 commented:
Hi Matheus,
I'm back with some updates. Over the past few days, I've done a lot of testing and found more bugs. We’d fix one, and others would show up — lol. But eventually, I figured it out, and now everything seems perfect.

I'm sharing the files with you as well so you can test them — I made a few improvements and also added a debug tool for the audio. I'm really excited about it.

In the archive, you'll find a .docx file with some details about the updates and changes, as well as the two modified files: main.py and utils.py. I didn't make a pull request yet because I thought you might want to test them first. If you're happy with the changes, you can update the PR yourself — or maybe you have a different idea.

I focused mostly on the transcribe function, but along the way I had to make some changes to the translate function as well.

Hope you like it!

[info.zip](https://github.com/user-attachments/files/21742377/info.zip)

MaKTaiL commented:
> Hi Matheus, I'm back with some updates. Over the past few days, I've done a lot of testing and found more bugs. We’d fix one, and others would show up — lol. But eventually, I figured it out, and now everything seems perfect.
> 
> I'm sharing the files with you as well so you can test them — I made a few improvements and also added a debug tool for the audio. I'm really excited about it.
> 
> In the archive, you'll find a .docx file with some details about the updates and changes, as well as the two modified files: main.py and utils.py. I didn't make a pull request yet because I thought you might want to test them first. If you're happy with the changes, you can update the PR yourself — or maybe you have a different idea.
> 
> I focused mostly on the transcribe function, but along the way I had to make some changes to the translate function as well.
> 
> Hope you like it!
> 
> [info.zip](https://github.com/user-attachments/files/21742377/info.zip)

Thank you. I'll try it on my side and I'll let you know.

angelitto2005 commented:
Don't test it yet, wait a bit longer.
Yesterday I tested all day and the script worked perfectly. After I sent it to you, guess what — I transcribed another movie and found another bug. It's not from the script, but from Gemini. I thought it was fixed, but apparently not.
Sometimes the transcription progress bar exceeds 100% due to invalid timestamps, like this:
Transcribing: |████████████████████████████████████████████████████████████████████████████████████████████| 306% (08:18:15,000/02:42:26,376) gemini-2.5-pro | Processing

I think I fixed it today, but I need a day or two for testing.
So let me test it first and I’ll get back to you with the fixed code.

angelitto2005 commented:
Matheus,
OMG, I finally managed to fix a bug that had been stressing me out for months. This one:

```
Transcribing: |██████████████░░░░░░░░░░░░░░░░| 49% (01:19:59,000/02:42:26,376) gemini-2.5-pro | Thinking \
Transcribing: |██████████████░░░░░░░░░░░░░░░░| 49% (01:19:59,000/02:42:26,376) gemini-2.5-pro | Thinking |
Transcribing: |██████████████░░░░░░░░░░░░░░░░| 49% (01:19:59,000/02:42:26,376) gemini-2.5-pro | Thinking /
Transcribing: |██████████████░░░░░░░░░░░░░░░░| 49% (01:19:59,000/02:42:26,376) gemini-2.5-pro | Thinking —
Transcribing: |██████████████░░░░░░░░░░░░░░░░| 49% (01:19:59,000/02:42:26,376) gemini-2.5-pro | Thinking \
Transcribing: |██████████████░░░░░░░░░░░░░░░░| 49% (01:19:59,000/02:42:26,376) gemini-2.5-pro | Thinking |
Transcribing: |██████████████░░░░░░░░░░░░░░░░| 49% (01:19:59,000/02:42:26,376) gemini-2.5-pro | Thinking /
Transcribing: |██████████████░░░░░░░░░░░░░░░░| 49% (01:19:59,000/02:42:26,376) gemini-2.5-pro | Thinking —
Transcribing: |██████████████░░░░░░░░░░░░░░░░| 49% (01:19:59,000/02:42:26,376) gemini-2.5-pro | Thinking \
Transcribing: |██████████████░░░░░░░░░░░░░░░░| 49% (01:19:59,000/02:42:26,376) gemini-2.5-pro | Thinking |
Transcribing: |██████████████░░░░░░░░░░░░░░░░| 49% (01:19:59,000/02:42:26,376) gemini-2.5-pro | Thinking /
Transcribing: |██████████████░░░░░░░░░░░░░░░░| 49% (01:19:59,000/02:42:26,376) gemini-2.5-pro

Transcribing: |██████████████░░░░░░░░░░░░░░░░| 49% (01:19:59,000/02:42:26,376) gemini-2.5-pro | Thinking —

Transcribing: |██████████████░░░░░░░░░░░░░░░░| 49% (01:19:59,000/02:42:26,376) gemini-2.5-pro | Thinking \

Transcribing: |██████████████░░░░░░░░░░░░░░░░| 49% (01:19:59,000/02:42:26,376) gemini-2.5-pro | Thinking |

Transcribing: |██████████████░░░░░░░░░░░░░░░░| 49% (01:19:59,000/02:42:26,376) gemini-2.5-pro | Thinking /

```

Now the console is clean and it shows only a limited number of lines. My settings are 20:

```
Resuming transcription from 06:31

Transcribing: |███████████████████████░░░░░░░| 77% (02:06:27,000/02:42:26,376) gemini-2.5-flash | Thinking /

Batch 3 thinking process saved to file.
Transcribing audio segment 36:16 to 46:31.
Batch 4 thinking process saved to file.
Transcribing audio segment 46:16 to 56:31.
Batch 5 thinking process saved to file.
Transcribing audio segment 56:16 to 66:31.
Batch 6 thinking process saved to file.
Transcribing audio segment 66:16 to 76:31.
Batch 7 thinking process saved to file.
Transcribing audio segment 76:16 to 86:31.
Batch 8 thinking process saved to file.
Transcribing audio segment 86:16 to 96:31.
Batch 9 thinking process saved to file.
Transcribing audio segment 96:16 to 106:31.
Batch 10 thinking process saved to file.
Transcribing audio segment 106:16 to 116:31.
Batch 11 thinking process saved to file.
Transcribing audio segment 116:16 to 126:31.
Batch 12 thinking process saved to file.
Transcribing audio segment 126:16 to 136:31.
```


You can set it in logger.py 

```
# Store the last progress bar state for message updates
_last_progress = None
_has_started = False
_previous_messages = []
_last_chunk_size = 0
MAX_LOG_LINES = 20  # <-- ADDED: A sensible limit for redrawing

```

angelitto2005 commented:
Hi Matheus,
news are coming... soon 😊
Just need a little more testing. Everything seems cool so far. Lots of improvements, especially in the Transcribe function.

But for now, I’ve made some edits to ffmpeg_utils.py for you to test it.

### Subject: Updates to the audio processing logic within the extract_audio_from_video function.

The primary goal of these changes was to improve the quality and robustness of the final MP3 file when the isolate_voice option is set to False.
Here are the detailed changes:

### 1. Behavior for isolate_voice=False (Modified):

    For 5.1/7.1 Audio: The downmix logic has been completely replaced.
        Before: A simple mathematical average of all channels was used (pan=mono|c0=(1/N)*c0 + ...). This often resulted in the dialogue being drowned out by music and sound effects from the surround channels.
        Now: An intelligent, weighted downmix based on industry standards has been implemented: pan=mono|c0<1.0*FC+0.707*FL+0.707*FR+0.5*LFE. This method prioritizes the dialogue channel (Front Center), adds context from the front channels (Left/Right), and avoids noise from the rear channels, resulting in much clearer dialogue.

    For All Audio Types (Stereo, 5.1, etc.):
        A 70% volume boost filter (volume=1.70) has been added. This is applied at the end of the filter chain to compensate for perceived volume loss during the downmix process and to deliver a more robust, normalized final audio file.

### 2. Behavior for isolate_voice=True (Unchanged):

    This functionality has been preserved exactly as it was, with no modifications.
        For 5.1/7.1 audio, it continues to use the strict method of extracting only the Front Center channel (pan=mono|c0=FC).
        For stereo audio, it continues to apply the standard downmix followed by voice-isolating filters (highpass, lowpass).

In summary, the "voice isolation" feature remains untouched, while the standard audio extraction mode has been significantly upgraded for superior dialogue clarity and a healthier volume level.

If you're wondering why there's an 80% volume increase, it's because the volume drops noticeably when downmixing from stereo or 5.1/7.1 to mono. With that 80% boost, it sounds like the original file.

And why is the final MP3 50MB? Because it has a bitrate of 64kbps — the exact same bitrate used by Pydub. So the quality of the full extracted MP3 is the same as the 10-minute segments sent to Gemini, with no loss in quality.

Anyway, anything over 50MB doesn't really matter, because Pydub re-compresses each audio segment to 64kbps anyway. I even tested with a 100MB file, but once I saw how Pydub compresses it, I stuck with the standard 50MB at 64kbps.

Now I can test the audio segments sent to Gemini for translation and transcription, thanks to the new debug audio function I implemented — and it works perfectly. In the next few days, I’ll also send you the updates I mentioned in the previous posts.

One more thing. I implemented this intelligent downmix because I ran into a bug. I extracted the audio from the movie My Dead Friend Zoe (from YTS), and with voice isolation enabled, I could only hear the voices — but not the music Rihanna sings at the beginning of the film. That part I think should also be transcribed.
If I disabled voice isolation, then I couldn't hear the voices at all — only the ambient sounds and background music. The track is AAC, though I'm not sure if that's relevant.

With this fix, I can now use voice isolation off and still successfully transcribe tracks like this that have issues.

Also, I found another bug: I came across a movie with 5.1 audio, but the center channel didn’t contain the voices — probably due to incorrect channel mapping. As a result, only ambient sounds were audible.

For problematic tracks like these, this intelligent downmix combined with the 80% volume increase works perfectly.

[ffmpeg_utils v2.zip](https://github.com/user-attachments/files/21824731/ffmpeg_utils.v2.zip)


Test it out when you’ve got time, and have a great week!

MaKTaiL commented:
> Hi Matheus, news are coming... soon 😊 Just need a little more testing. Everything seems cool so far. Lots of improvements, especially in the Transcribe function.
> 
> But for now, I’ve made some edits to ffmpeg_utils.py for you to test it.
> 
> ### Subject: Updates to the audio processing logic within the extract_audio_from_video function.
> The primary goal of these changes was to improve the quality and robustness of the final MP3 file when the isolate_voice option is set to False. Here are the detailed changes:
> 
> ### 1. Behavior for isolate_voice=False (Modified):
> ```
> For 5.1/7.1 Audio: The downmix logic has been completely replaced.
>     Before: A simple mathematical average of all channels was used (pan=mono|c0=(1/N)*c0 + ...). This often resulted in the dialogue being drowned out by music and sound effects from the surround channels.
>     Now: An intelligent, weighted downmix based on industry standards has been implemented: pan=mono|c0<1.0*FC+0.707*FL+0.707*FR+0.5*LFE. This method prioritizes the dialogue channel (Front Center), adds context from the front channels (Left/Right), and avoids noise from the rear channels, resulting in much clearer dialogue.
> 
> For All Audio Types (Stereo, 5.1, etc.):
>     A 70% volume boost filter (volume=1.70) has been added. This is applied at the end of the filter chain to compensate for perceived volume loss during the downmix process and to deliver a more robust, normalized final audio file.
> ```
> 
> ### 2. Behavior for isolate_voice=True (Unchanged):
> ```
> This functionality has been preserved exactly as it was, with no modifications.
>     For 5.1/7.1 audio, it continues to use the strict method of extracting only the Front Center channel (pan=mono|c0=FC).
>     For stereo audio, it continues to apply the standard downmix followed by voice-isolating filters (highpass, lowpass).
> ```
> 
> In summary, the "voice isolation" feature remains untouched, while the standard audio extraction mode has been significantly upgraded for superior dialogue clarity and a healthier volume level.
> 
> If you're wondering why there's an 80% volume increase, it's because the volume drops noticeably when downmixing from stereo or 5.1/7.1 to mono. With that 80% boost, it sounds like the original file.
> 
> And why is the final MP3 50MB? Because it has a bitrate of 64kbps — the exact same bitrate used by Pydub. So the quality of the full extracted MP3 is the same as the 10-minute segments sent to Gemini, with no loss in quality.
> 
> Anyway, anything over 50MB doesn't really matter, because Pydub re-compresses each audio segment to 64kbps anyway. I even tested with a 100MB file, but once I saw how Pydub compresses it, I stuck with the standard 50MB at 64kbps.
> 
> Now I can test the audio segments sent to Gemini for translation and transcription, thanks to the new debug audio function I implemented — and it works perfectly. In the next few days, I’ll also send you the updates I mentioned in the previous posts.
> 
> One more thing. I implemented this intelligent downmix because I ran into a bug. I extracted the audio from the movie My Dead Friend Zoe (from YTS), and with voice isolation enabled, I could only hear the voices — but not the music Rihanna sings at the beginning of the film. That part I think should also be transcribed. If I disabled voice isolation, then I couldn't hear the voices at all — only the ambient sounds and background music. The track is AAC, though I'm not sure if that's relevant.
> 
> With this fix, I can now use voice isolation off and still successfully transcribe tracks like this that have issues.
> 
> Also, I found another bug: I came across a movie with 5.1 audio, but the center channel didn’t contain the voices — probably due to incorrect channel mapping. As a result, only ambient sounds were audible.
> 
> For problematic tracks like these, this intelligent downmix combined with the 80% volume increase works perfectly.
> 
> [ffmpeg_utils v2.zip](https://github.com/user-attachments/files/21824731/ffmpeg_utils.v2.zip)
> 
> Test it out when you’ve got time, and have a great week!

The reason I took out the 80% volume increase was because Gemini was adding an exclamation point ( ! ) at the end of all sentences when I was testing transcription. Gemini thought everyone was screaming because of the volume increase and always added it even when I asked it not to.

angelitto2005 commented:
> The reason I took out the 80% volume increase was because Gemini was adding an exclamation point ( ! ) at the end of all sentences when I was testing transcription. Gemini thought everyone was screaming because of the volume increase and always added it even when I asked it not to.

 I've never encountered this kind of behavior during my testing. Maybe it was a bug in Gemini back then, and it's been fixed now?
 Just to clarify: the 80% volume boost applies only when Voice Isolation is OFF. It doesn't affect the Voice Isolation ON setting. And even with that 80% increase, the audio volume only matches the original file. It’s not actually louder.
 On my end, with VI OFF, after applying the 80% boost, I noticed a clear improvement in transcription. Many sentences that previously weren’t being transcribed are now showing up correctly.
Let me know if you encounter that bug again with the file I sent you.

angelitto2005 commented:
Hi Matheus,
just following up with some updates.
There are some new and cool enhancements.  I hope you’ll have time to test them out, and if you like them, feel free to implement them into the script.
There are quite a few changes, and honestly, I’m not sure I remember them all! 😁

Here's the list:

### 1. File: ffmpeg_utils.py

> 1. The default value for the `target_mb` parameter in the `extract_audio_from_video` function was increased from 20 to 50 MB.
> 2. The audio processing logic in `extract_audio_from_video` was completely restructured to differently handle the case where voice isolation (`isolate_voice`) is disabled.
> 3. A new intelligent downmix rule was added for 5.1/7.1 audio when `isolate_voice` is `False `to better preserve dialogue (pan=mono|c0<1.0*FC+0.707*FL+0.707*FR+0.5*LFE).
> 4. An ffmpeg filter was added to apply a 80% volume boost (`volume=1.80`) for all cases where `isolate_voice` is `False`.
> 5. The informational messages displayed in the console were updated to include ANSI color codes, improving readability.

### 2. File: logger.py

> 1. A new function, `log_detailed_error`, was added to log detailed errors to the log file without displaying duplicate messages in the console.
> 2. The `clear_session_logs` function was added to reset logs at the start of a new session (one that is not a resume).
> 3. The `progress_bar` function was extensively refactored to fix a bug that caused console lines to be overwritten, by correctly calculating the wrapped text height with the `wrapped_lines` function.
> 4. A limitation was added to the `progress_bar` to ensure the displayed progress never exceeds 100%.
> 5. The color of the progress bar was changed from green/blue to green/yellow.
> 6. A new variable, `_all_messages`, was introduced to store a complete history of messages for the log file, while `_previous_messages` (for the console) is now limited to the last 20 messages.
> 7. The `save_logs_to_file` function was modified to accept a `resume_mode` parameter.
> 8. In `save_logs_to_file`, the logic was changed to update the existing progress bar line in the file instead of appending a new one on each save.
> 9. The `save_thoughts_to_file` function was modified to receive `batch_number` as a parameter and to correctly handle appending content in resume mode.
> 10. New helper functions were added: `wrapped_lines`, `_find_and_replace_progress_bar_in_file`, and `_format_seconds_to_timestamp`.
> 11. The color for the progress function was changed from blue to yellow.

### 3. File: utils.py

> 1. The `convert_timedelta_to_timestamp` function was fixed to use `total_seconds()` instead of `.seconds`, allowing for the correct conversion of durations longer than one minute.
> 2. `Convert_timedelta_to_timestamp` was modified to also be able to display milliseconds.
> 3. The `convert_timestamp_to_timedelta` function was improved to parse multiple time formats (e.g., `MM:SS` and `HH:MM:SS`).

### 4. File: main.py

> 1. A new parameter, `debug_audio: bool = False`, was added to the GeminiSRTTranslator class to enable the saving of processed audio chunks.
> 2. The resume logic (`_check_saved_progress` and `_check_saved_transcribe_progress`) was updated to read and restore the `batch_number` from the progress file.
> 3. The progress-saving functions (`_save_progress`, `_save_transcribe_progress`) now also save the current `batch_number`.
> 4. Robust API key validation logic was added at the start of the `translate `and `transcribe `functions, which can switch to a secondary key if the primary one is invalid.
> 5. In both the `translate `and `transcribe `methods, functionality was added to automatically save processed audio segments and their corresponding subtitles to a "Debug" folder if `debug_audio` is `True`.
> 6. The error handling loop in `translate `and `transcribe `was refactored: the consecutive error counter is now incremented more intelligently, it handles the "504 Gateway Timeout" error, and it uses the new `log_detailed_error` function.
> 7. In `transcribe`, an audio chunk overlap system was added to improve transcription continuity.
> 8. Multiple strict validations for timestamps returned by the API during transcription were added (checking for missing values, end time before start time, values outside the expected range, etc.).
> 9. The interrupt handler (`handle_interrupt`) was updated to delete the last unprocessed debug audio file on exit.
> 10. The transcription resume logic (`_check_saved_transcribe_progress`) was improved to automatically remove the last potentially incomplete subtitle on resume, ensuring it gets re-transcribed correctly for seamless continuity. (This fix rocks.)

I've tested them a lot over the past week to make sure everything runs like clockwork, and I think you'll be pleased with the results.

Cheers!

[gst v20.zip](https://github.com/user-attachments/files/21949767/gst.v20.zip)

MaKTaiL commented:
Hi @angelitto2005 , are the changes in v20 included in the PR or are they separate for the moment?

angelitto2005 commented:
> Hi @angelitto2005 , are the changes in v20 included in the PR or are they separate for the moment?

I started from the changes in the Pull Request and continued progressing with enhancements and fixes. The final version is v20. So, the changes from the PR are included, or rather, improved in v20. Therefore, please test the v20 archive directly.

I wasn't sure how to proceed, whether to close the PR and open a new one, or if you'd prefer to adjust things yourself as you see fit.

angelitto2005 commented:
I'm really happy with the results. Two friends and I tested it thoroughly, especially using 2.5 Pro, and everything works well so far.

I'm using at the moment google-genai 1.31.0 and json_repair 0.48.0

2.5 Pro runs very slowly during the day (it only works without errors at night), but those daytime errors actually helped me create fixes for all sorts of unexpected issues and bugs. So, in every bad situation, there's also something good 😊

MaKTaiL commented:
Alright, I'll do final tests today. I think I'll just update this PR with the changes.

angelitto2005 commented:
> Alright, I'll do final tests today. I think I'll just update this PR with the changes.

Ok. Thanks!
